datalad/metadata/aggregate.py

```diff
@@ -11,32 +11,146 @@
 
 __docformat__ = 'restructuredtext'
 
-import os
-from os.path import join as opj, exists, relpath, dirname
+from os.path import join as opj
+from os.path import dirname
+from os.path import relpath
 from datalad.interface.base import Interface
+from datalad.interface.utils import eval_results
+from datalad.interface.utils import discover_dataset_trace_to_targets
+from datalad.interface.save import Save
 from datalad.interface.base import build_doc
-from datalad.interface.utils import handle_dirty_dataset
 from datalad.interface.common_opts import recursion_limit, recursion_flag
-from datalad.interface.common_opts import if_dirty_opt
 from datalad.interface.common_opts import nosave_opt
-from datalad.utils import with_pathsep as _with_sep
-from datalad.distribution.dataset import datasetmethod, EnsureDataset, \
-    Dataset, require_dataset
-from ..support.param import Parameter
-from ..support.constraints import EnsureNone
-from datalad.support.exceptions import CommandError
-from ..log import lgr
-from . import get_metadata, metadata_filename, metadata_basepath, is_implicit_metadata
+from datalad.interface.results import get_status_dict
+from datalad.distribution.dataset import Dataset
+from datalad.metadata.metadata import Metadata
+from datalad.metadata.metadata import agginfo_relpath
+from datalad.metadata.metadata import _load_json_object
+from datalad.distribution.dataset import datasetmethod, EnsureDataset, require_dataset
+from datalad.support.param import Parameter
+from datalad.support.constraints import EnsureStr
+from datalad.support.constraints import EnsureNone
+from datalad.log import lgr
 from datalad.support.json_py import dump as jsondump
 
 
-def _store_json(ds, path, meta):
-    if not exists(path):
-        os.makedirs(path)
-    fname = opj(path, metadata_filename)
-    jsondump(meta, fname)
-    # stage potential changes
-    ds.repo.add(fname, git=True)
+def _adj2subbranches(base, adj):
+    # given a set of parent-child mapping, compute a mapping of each parent
+    # to all its (grand)children of any depth level
+    branches = dict(adj)
+    # from bottom up
+    for ds in sorted(adj, reverse=True):
+        subbranch = []
+        for sub in branches[ds]:
+            subbranch.append(sub)
+            subbranch.extend(branches.get(sub, []))
+        branches[ds] = subbranch
+    return branches
```

Comment:
It refers to the branches of the dataset tree. Honestly I don't care about this name, it is use in this place in less than a handful places -- a random string is fine with me.